### DISCUSSION TITLE
format() and ButtonGroupColumn() not working as expected

### DISCUSSION THREAD

**Discussion by faridsa** [rappasoft/laravel-livewire-tables#780] [Q&A]
Hello I'm using v.2.7.0 and trying 2 features that are frustrating me a lot, because results are not as expected.

1) in v1.x I can make a column like this 
```
Column::make('Full Name', 'fullname')
                ->searchable(),
```
where fullname is just a function in my model as
```
public function getFullnameAttribute()
    {
        return "{$this->name} {$this->surname}";
    }
```
And it returns the expected value.

Now in v2 it fires and SQL error
```SQLSTATE[42S22]: Column not found: 1054 Unknown column 'estimates.fullname' in 'field list' (SQL: select `estimates`.`fullname` as `fullname`, ```

So trying to obtain it by other method, I went with format(), in this way (as documentation states)
```
Column::make('Full Name')
                ->format(
                    fn ($value, $row, Column $column) => $row->name . ' ' . $row->surname
                )->searchable(),
```
And now it fires another SQL error
```SQLSTATE[42S22]: Column not found: 1054 Unknown column 'estimates.full_name' in 'field list' (SQL: select `estimates`.`full_name` as `full_name`, ```

So it makes me clear that any column name must match a database field, but if I use the following variation:
```
Column::make('Full Name', 'name')
                ->format(
                    fn ($value, $row, Column $column) => $row->name . ' ' . $row->surname
                )->searchable(),
```
I have no errors now, but **Full Name** column only returns name

-------------

2) In v1  I can make an empty column to put there my action buttons using just:
`Column::make('Actions', ''),`

But now in v2, that fires and SQL error like:
`SQLSTATE[42S22]: Column not found: 1054 Unknown column 'advises.actions' in 'field list' `

So checking in the docs I found a function that should be exactly what I need **_ButtonGroupColumn()_**

```
ButtonGroupColumn::make('Actions')
    ->attributes(function($row) {
        return [
            'class' => 'space-x-2',
        ];
    })
    ->buttons([
        LinkColumn::make('View') // make() has no effect in this case but needs to be set anyway
            ->title(fn($row) => 'View ' . $row->name)
            ->location(fn($row) => route('estimates.show', $row))/*, this comma fires and error so I suppressed it*/
            ->attributes(function($row) {
                return [
                    'class' => 'underline text-blue-500 hover:no-underline',
                ];
            }),
        LinkColumn::make('Edit')
            ->title(fn($row) => 'Edit ' . $row->name)
            ->location(fn($row) => route('estimates.edit', $row))/*, this comma fires and error so I suppressed it*/
            ->attributes(function($row) {
                return [
                    'target' => '_blank',
                    'class' => 'underline text-blue-500 hover:no-underline',
                ];
            }),
    ]),
```
But now the column **Actions** simply doesn't renders

**Comment by inmanturbo** [comment]
Hello @faridsa, have you tried the label method? 

https://rappasoft.com/docs/laravel-livewire-tables/v2/columns/available-methods#content-labels

like this:
```php
Column::make('Full Name')
    ->label(
        fn($row, Column $column) => $row->full_name
    ),
```
also may need on the model:
```php
protected $appends = [ 'full_name' ];
```

For your actions you can also try:
```php
Column::make('Actions')
    // Note: The view() method is reserved for columns that have a field
    ->label(
        fn($row, Column $column) => view('my.actions.view')->withRow($row)
    ),
```

**Reply by WojciechRhdt** [reply]
It is possible to sort a table by a column using the label method?

**Comment by inmanturbo** [comment]
Also, are you using the  column selection dropdown next to the per page dropdown in upper right above table, and do you have another table in your project with an `actions` column? Columns simply not rendering or being visible could be related to [this bug](https://github.com/rappasoft/laravel-livewire-tables/issues/760)

**Comment by xitude** [comment]
I'm also struggling to do some really basic stuff since upgrading. I'm starting to wonder if this version is ready to be the default pulled branch?

**Reply by giles247** [reply]
Not sure if it helps, but I've found only way to get certain values to work using, format, label, custom view etc... is to create a hidden column earlier for that field value e.g.

```
            Column::make('Name', 'name')
                ->hideIf(true), 
```

It's messy as hell but I've not found a better way so far. In my view there needs to be a simpler way to make all model fields (and those for related tables) either available or selectable for use in format, label, custom views etc... I'm also struggling with this version :(

**Comment by Humfurie** [comment]
Is there any update on this?
